resources/tests/test_resources_mutations.py

```diff
@@ -0,0 +1,262 @@
+        graphql_url="/graphql_v2/",
+        user=superuser,
+    )
+
+    assert Berth.objects.count() == 0
+
+
+def test_delete_berth_staff_user(staff_user, berth):
+    variables = {
+        "id": to_global_id("BerthNode", str(berth.id)),
+    }
+
+    assert Berth.objects.count() == 1
+
+    executed = client.execute(
+        query=DELETE_BERTH_MUTATION,
+        variables=variables,
+        graphql_url="/graphql_v2/",
+        user=staff_user,
+    )
+
+    assert Berth.objects.count() == 1
+    assert_not_enough_permissions(executed)
+
+
+def test_delete_berth_no_user(berth):
+    variables = {
+        "id": to_global_id("BerthNode", str(berth.id)),
+    }
+
+    assert Berth.objects.count() == 1
+
+    executed = client.execute(
+        query=DELETE_BERTH_MUTATION,
+        variables=variables,
+        graphql_url="/graphql_v2/",
+        user=None,
+    )
+
+    assert Berth.objects.count() == 1
+    assert_not_enough_permissions(executed)
+
+
+def test_delete_berth_inexistent_berth(superuser):
+    variables = {
+        "id": to_global_id("BerthNode", uuid.uuid4()),
+    }
+
+    executed = client.execute(
+        query=DELETE_BERTH_MUTATION,
+        variables=variables,
+        graphql_url="/graphql_v2/",
+        user=superuser,
+    )
+
+    assert_doesnt_exist("Berth", executed)
+
+
+UPDATE_BERTH_MUTATION = """
+    mutation UpdateBerth($input: UpdateBerthMutationInput!) {
+      updateBerth(input: $input) {
+        berth {
+          id
+          number
+          comment
+          pier {
+            id
+          }
+          berthType {
+            id
+          }
+        }
+      }
+    }
+"""
+
+
+def test_update_berth(berth, pier, berth_type, superuser):
+    global_id = to_global_id("BerthNode", str(berth.id))
+    pier_id = to_global_id("PierNode", str(pier.id))
+    berth_type_id = to_global_id("BerthTypeNode", str(berth_type.id))
+
+    variables = {
+        "id": global_id,
+        "number": "666",
+        "comment": "foobar",
+        "pierId": pier_id,
+        "berthTypeId": berth_type_id,
+    }
+
+    assert Berth.objects.count() == 1
+
+    executed = client.execute(
+        query=UPDATE_BERTH_MUTATION,
+        variables=variables,
+        graphql_url="/graphql_v2/",
+        user=superuser,
+    )
+
+    assert Berth.objects.count() == 1
+    assert executed["data"]["updateBerth"]["berth"]["id"] == global_id
+    assert executed["data"]["updateBerth"]["berth"]["comment"] == "foobar"
+    assert executed["data"]["updateBerth"]["berth"]["number"] == "666"
+    assert executed["data"]["updateBerth"]["berth"]["pier"]["id"] == pier_id
+    assert executed["data"]["updateBerth"]["berth"]["berthType"]["id"] == berth_type_id
+
+
+def test_update_berth_no_id(berth, pier, berth_type, superuser):
+    pier_id = to_global_id("PierNode", str(pier.id))
+    berth_type_id = to_global_id("BerthTypeNode", str(berth_type.id))
+
+    variables = {
+        "number": "666",
+        "comment": "foobar",
```

Comment:
also you could skip this "comment" field in most test cases (maybe keep it in one or two to check that it exists at all), as it is not a required field for the model, would have made your life 0.1% easier 😅  no need to delete now, of course, since you already have written all these lines.